docs/design/data-modeling.md

```diff
@@ -159,3 +172,8 @@ This technique enables a simple range query to iterate every object
 with a target role. While you can't yet do prefix queries within
 TigerBeetle, you could do a prefix query in an external database to
 resolve `id`s and pass the resolved `id`s into TigerBeetle.
+
+#### Random Identifier
+
+Random identifiers are not recommended – they are incompatible with some LSM optimizations.
```

Comment:
They are not incompatible (they still work), but they don't take advantage of all of TigerBeetle's optimizations.

Reply:
> resulting in +/- 5-10% reduction in performance.

Some real numbers from `main`, on my laptop:
35s sequential vs 43s random on tmpfs
52s sequential vs 60s random on nvme